bgpd/bgp_updgrp_packet.c

```diff
@@ -763,7 +763,13 @@ subgroup_update_packet (struct update_subgroup *subgrp)
 
 	  if (rn->prn)
 	    prd = (struct prefix_rd *) &rn->prn->p;
-          tag = bgp_adv_label(rn, binfo, peer, afi, safi);
+
+          if (safi == SAFI_LABELED_UNICAST)
+            tag = bgp_adv_label(rn, binfo, peer, afi, safi);
```

Comment:
If you are suggesting we call bgp_adv_label for all SAFIs and do the correct thing there based on the safi passed in, I have no objections to changing it in that way. If you mean something other than that, please explain.